modules/system/assets/js/framework.js

```diff
@@ -306,7 +319,9 @@ if (window.jQuery.request !== undefined) {
         context.error = requestOptions.error
         context.complete = requestOptions.complete
         requestOptions = $.extend(requestOptions, options)
-        requestOptions.data = data.join('&')
+        requestOptions.data = sendFiles ? formdata : data
```

Comment:
Perfect, thanks for the explanation! Have you tested this latest refactor thoroughly with and without using fileuploads? Does the old method of uploading files over ajax used by the mediamanager and fileupload widgets still work? Is it worthwhile to also convert those widgets to use this new functionality?